src/fondant/core/manifest.py

```diff
@@ -220,41 +195,76 @@ def cache_key(self) -> str:
         return self.metadata["cache_key"]
 
     @property
-    def index(self) -> Index:
-        return Index(self._specification["index"], base_path=self.base_path)
-
-    @property
-    def subsets(self) -> t.Mapping[str, Subset]:
+    def fields(self) -> t.Mapping[str, Field]:
         """The subsets of the manifest as an immutable mapping."""
+        # e.g. ('images', {'location': '/component1', 'type': 'binary'})
```

Comment:
This comment is confusing to me. It shows a tuple, while we're returning a mapping.